server/src/test/java/com/google/coffeehouse/storagehandler/StorageHandlerTest.java

```diff
@@ -0,0 +1,246 @@
+  }
+
+  public void insertBookWithNullAuthor() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithNullAuthor")
+        .set("isbn")
+        .to("9780060959470")
+        .set("title")
+        .to("all about love: new visions")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithNullIsbn() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithNullIsbn")
+        .set("author")
+        .to("bell hooks")
+        .set("title")
+        .to("all about love: new visions")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithNullAuthorAndNullIsbn() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithNullAuthorAndNullIsbn")
+        .set("title")
+        .to("anonymous book about love")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithEmptyAuthor() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithEmptyAuthor")
+        .set("author")
+        .to("")
+        .set("isbn")
+        .to("9780060959470")
+        .set("title")
+        .to("all about love: new visions")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithEmptyIsbn() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithEmptyIsbn")
+        .set("author")
+        .to("bell hooks")
+        .set("isbn")
+        .to("")
+        .set("title")
+        .to("all about love: new visions")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithEmptyAuthorAndEmptyIsbn() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithEmptyAuthorAndEmptyIsbn")
+        .set("author")
+        .to("")
+        .set("isbn")
+        .to("")
+        .set("title")
+        .to("anonymous book about love")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithEmptyAuthorAndNullIsbn() {
+    List<Mutation> mutations = new ArrayList<>();
+    mutations.add(
+      Mutation.newInsertOrUpdateBuilder("Books")
+        .set("bookId")
+        .to("bookWithEmptyAuthorAndNullIsbn")
+        .set("author")
+        .to("")
+        .set("title")
+        .to("anonymous book about love")
+        .build());
+    dbClient.write(mutations);
+  }
+
+  public void insertBookWithNullAuthorAndEmptyIsbn() {
```

Comment:
It seems like a lot of these functions aren't used anywhere. Maybe take them out for now, or move them to the test helper